frontend/database.go

```diff
@@ -17,30 +17,50 @@ const (
 // DBClient defines the needed values to perform CRUD operations against the async DB
 type DBClient struct {
 	client *azcosmos.Client
-	DBName string
-	DBUrl  string
+	config *DBConfig
+}
+
+// DBConfig stores database and client configuration data
+type DBConfig struct {
+	DBName        string
+	DBUrl         string
+	Cred          *azidentity.DefaultAzureCredential
+	ClientOptions *azidentity.DefaultAzureCredentialOptions
+}
+
+// NewDatabaseConfig configures database configuration values for access
+func NewDatabaseConfig() *DBConfig {
+	opt := &azidentity.DefaultAzureCredentialOptions{}
+	cred, _ := azidentity.NewDefaultAzureCredential(opt)
+	c := &DBConfig{
+		DBName:        os.Getenv("DB_NAME"),
+		DBUrl:         os.Getenv("DB_URL"),
+		Cred:          cred,
+		ClientOptions: opt,
+	}
+
+	return c
 }
 
 // NewDatabaseClient instanstiates a Cosmos DatabaseClient targeting Frontends async DB
-func NewDatabaseClient() *DBClient {
-	cred, _ := azidentity.NewDefaultAzureCredential(nil)
+func NewDatabaseClient(config *DBConfig) *DBClient {
 	d := &DBClient{
-		DBName: os.Getenv("DB_NAME"),
-		DBUrl:  os.Getenv("DB_URL"),
+		config: config,
 	}
-	client, _ := azcosmos.NewClient(d.DBUrl, cred, nil)
-	d.client = client
 
+	client, _ := azcosmos.NewClient(d.config.DBUrl, d.config.Cred, nil)
```

Comment:
why not check this error?

Reply:
good call, think i missed this because  client is defined in the struct which would't allow for error handling further up. I can move this out like how database config is done now so we can have error handling

```
func NewFrontend(logger *slog.Logger, listener net.Listener, emitter metrics.Emitter, dbConfig *DBConfig) *Frontend {
	f := &Frontend{
		logger:   logger,
		listener: listener,
		metrics:  emitter,
		server: http.Server{
			ErrorLog: slog.NewLogLogger(logger.Handler(), slog.LevelError),
			BaseContext: func(net.Listener) context.Context {
				return ContextWithLogger(context.Background(), logger)
			},
		},
		cache:    *NewCache(),
		dbClient: *NewDatabaseClient(dbConfig),  <---
		done:     make(chan struct{}),
	}
```

Reply:
ok added the error check and moved the client setup to main so it can be passed to Frontend and includes the DB config with it. Validated changes still work in dev environment.